loader/loader.c

```diff
@@ -2763,6 +2792,75 @@ static void VerifyAllMetaLayers(struct loader_instance *inst, struct loader_laye
     }
 }
 
+// If the current working directory matches any app_key_path of the layers, remove all other override layers.
+// Otherwise if no matching app_key was found, remove all but the global override layer, which has no app_key_path.
+static void RemoveAllNonValidOverrideLayers(struct loader_instance *inst, struct loader_layer_list *instance_layers) {
+    if (instance_layers == NULL) {
+        return;
+    }
+
+    char cur_path[MAX_STRING_SIZE];
+    char *ret = loader_platform_executable_path(cur_path);
```

Comment:
In the current code, the issue is within `loader_platform_executable_path()`, but if you make the change I suggested about the fixed size buffer, it'll matter here instead. Don't just hard code 1024 in the function call. It makes it much easier to mess thing up down the line when making changes. My preferred method is to use `sizeof(cur_path)`, though `MAX_STRING_SIZE` would also be reasonable.

Reply:
I have addressed all the issues and pushed up the fixes.
Having a size function parameter is a lot better and removes icky constants I have to admit.